File: C++/A/686A-freeicecream.cpp
#include <bits/stdc++.h>
using namespace std;

int main()
{
    long long n, x, d;
    cin >> n >> x;
    long long stock = x, distress = 0;
    char transaction;
    while (n--)
    {
        cin >> transaction >> d;
        if (transaction == '+')
            stock += d;
        else
        {
            (stock >= d) ? stock -= d : distress++;
        }
    }
    cout << stock << ' ' << distress;
}

File: Python/686A-freeicecream.py
n, x = map(int, input().split())
stock = x
distress = 0

for _ in range(n):
    transaction, d = input().split()
    d = int(d)

    if transaction == '+':
        stock += d
    else:
        if stock >= d:
            stock -= d
        else:
            distress += 1

print(stock, distress)
